relays/substrate-client/src/chain.rs

```diff
@@ -38,6 +44,13 @@ pub trait Chain: ChainBase {
 	type SignedBlock: Member + Serialize + DeserializeOwned + BlockWithJustification;
 	/// The aggregated `Call` type.
 	type Call: Dispatchable + Debug;
+	/// Balance of an account in native tokens.
+	type NativeBalance: Parameter + Member + DeserializeOwned + Clone + Copy + CheckedSub + PartialOrd + Zero;
+
+	/// Return runtime storage key for that is storing `pallet_balances::AccountData` of given account.
+	///
+	/// Would panic if chain uses different format to store account data.
```

Comment:
Why would this panic?

Reply:
If chain stores something other than `pallet_balances::AccountData`, then this fn should panic. But actually, maybe it worth additional trait - something like `ChainWithBalancesAccountData`. Will change that

Reply:
Added `ChainWithBalances: Chain`